File: ProyectoAsisTrack/src/main/java/com/example/ProyectoAsisTrack/Model/Asistencia.java
package com.example.ProyectoAsisTrack.Model;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDate;

@Entity
@Table(name = "asistencia")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Asistencia {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private LocalDate fecha;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private EstadoAsistencia estado;

    @ManyToOne
    @JoinColumn(name = "estudiante_id")
    private Estudiantes estudiante;

    @ManyToOne
    @JoinColumn(name = "asignatura_id")
    private Asignaturas asignatura;
}

File: app/src/main/java/com/example/proyectoasistrack/data/model/Asistencia.kt
package com.example.proyectoasistrack.data.model

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "asistencia")
data class Asistencia(
    @PrimaryKey(autoGenerate = true) val id: Int = 0,
    val estudiante: String,
    val fecha: String,
    val estado: String
)

